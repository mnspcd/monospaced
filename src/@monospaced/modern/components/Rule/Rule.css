.mdrn .Rule {
  background-color: var(--color-grey-2);
  border: 0;
  color: var(--color-grey-2);
  height: calc(1rem / 16);
  margin-bottom: 0;
  margin-top: 0;
}

.mdrn .Rule--onDarkBackground {
  background-color: var(--color-grey-7);
  color: var(--color-grey-7);
}
