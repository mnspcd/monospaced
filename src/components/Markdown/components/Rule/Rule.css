.Rule {
  background-color: var(--mnspcd-color-grey-2);
  border: 0;
  color: var(--mnspcd-color-grey-2);
  height: calc(1rem / 16);
  margin-bottom: 0;
  margin-top: 0;
}

.Rule--onDarkBackground {
  background-color: var(--mnspcd-color-grey-8);
  color: var(--mnspcd-color-grey-8);
}
