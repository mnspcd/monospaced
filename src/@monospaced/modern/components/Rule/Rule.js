import classNames from "classnames";
import React from "react";

import { SurfaceConsumer } from "../Surface";

import "./Rule.css";

/**
 * Use `Rule` to display a horizontal rule.
 */
const Rule = () => (
  <SurfaceConsumer>
    {surfaceBackground => (
      <hr
        className={classNames({
          Rule: true,
          [`Rule--onDarkBackground`]: surfaceBackground === "dark",
        })}
      />
    )}
  </SurfaceConsumer>
);

export default Rule;
