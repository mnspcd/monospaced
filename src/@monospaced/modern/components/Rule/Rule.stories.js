import { select } from "@storybook/addon-knobs";
import React from "react";

import Surface from "../Surface";

import Rule from "./Rule";

export default {
  component: Rule,
  title: "Components/Rule",
};

export const basic = () => <Rule />;

export const onDarkBackground = () => (
  <Surface backgroundColor="grey-dark">
    <div style={{ padding: "1em" }}>
      <Rule />
    </div>
  </Surface>
);

export const knobs = () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
  >
    <Rule />
  </Surface>
);

knobs.story = { parameters: { docs: { disable: true } } };
