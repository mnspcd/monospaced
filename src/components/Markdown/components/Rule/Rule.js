import React from "react";
import classNames from "classnames";

import { SurfaceConsumer } from "../../../Surface";
import "./Rule.css";

const Rule = () => (
  <SurfaceConsumer>
    {surfaceBackground => (
      <hr
        className={classNames({
          Rule: true,
          [`Rule--onDarkBackground`]: surfaceBackground === "dark",
        })}
      />
    )}
  </SurfaceConsumer>
);

export default Rule;
