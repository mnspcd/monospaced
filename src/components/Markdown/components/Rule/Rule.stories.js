import React from "react";
import { select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Surface from "../../../Surface";
import Rule from "./Rule";

storiesOf("Components/Markdown/Rule", module).add("Component", () => (
  <Surface
    backgroundColor={select(
      "backgroundColor",
      Surface.backgroundColors,
      "white",
      "Surface",
    )}
    padding={text("padding", "1em", "Surface")}
  >
    <Rule />
  </Surface>
));
