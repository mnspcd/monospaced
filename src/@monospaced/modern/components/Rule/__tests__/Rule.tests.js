/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Rule from "..";

describe("Rule component", () => {
  it("should render correctly", () => {
    const { container } = render(<Rule />);

    expect(container).toMatchSnapshot();
  });
});
