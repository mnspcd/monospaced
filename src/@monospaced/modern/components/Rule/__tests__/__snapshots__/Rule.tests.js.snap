// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rule component should render correctly 1`] = `
<div>
  <hr
    class="Rule"
  />
</div>
`;
